poly-commitment/src/lagrange_cache.rs

```diff
@@ -0,0 +1,201 @@
+            LagrangeCache::Remote(cache) => cache.load_lagrange_basis_from_cache(g, domain),
+        }
+    }
+
+    fn cache_lagrange_basis(
+        &self,
+        g: &Vec<G>,
+        domain: &D<G::ScalarField>,
+        basis: &Vec<PolyComm<G>>,
+    ) {
+        match self {
+            LagrangeCache::Local(cache) => cache.cache_lagrange_basis(g, domain, basis),
+            LagrangeCache::Remote(cache) => cache.cache_lagrange_basis(g, domain, basis),
+        }
+    }
+}
+
+pub trait LagrangeCacheTrait<G: CommitmentCurve> {
+    type CacheKey;
+
+    fn lagrange_basis_cache_key(&self, g: &Vec<G>, domain: &D<G::ScalarField>) -> Self::CacheKey;
+
+    fn load_lagrange_basis_from_cache(
+        &self,
+        g: &Vec<G>,
+        domain: &D<G::ScalarField>,
+    ) -> Option<Vec<PolyComm<G>>>;
+
+    fn cache_lagrange_basis(
+        &self,
+        g: &Vec<G>,
+        domain: &D<G::ScalarField>,
+        basis: &Vec<PolyComm<G>>,
+    );
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FileCache<G> {
+    cache_dir: PathBuf,
+    point_type: PhantomData<G>,
+}
+
+impl<G> FileCache<G> {
+    pub fn new(cache_dir: PathBuf) -> Self {
+        if !cache_dir.exists() {
+            fs::create_dir_all(&cache_dir).unwrap();
+        }
+        FileCache {
+            cache_dir,
+            point_type: PhantomData,
+        }
+    }
+}
+
+impl<G: CommitmentCurve> LagrangeCacheTrait<G> for FileCache<G> {
+    type CacheKey = PathBuf;
+
+    fn lagrange_basis_cache_key(&self, g: &Vec<G>, domain: &D<G::ScalarField>) -> Self::CacheKey {
+        let mut hasher = DefaultHasher::new();
+        g.hash(&mut hasher);
+        domain.size.hash(&mut hasher);
+        self.cache_dir
+            .clone()
+            .join(format!("lagrange_basis_{:}", hasher.finish()))
+    }
+
+    fn load_lagrange_basis_from_cache(
+        &self,
+        g: &Vec<G>,
+        domain: &D<G::ScalarField>,
+    ) -> Option<Vec<PolyComm<G>>> {
+        let cache_key = self.lagrange_basis_cache_key(g, domain);
+        if Path::exists(&cache_key) {
+            let f = File::open(cache_key.clone()).expect(&format!(
+                "Missing lagrange basis cache file {:?}",
+                cache_key
+            ));
+            let basis = rmp_serde::decode::from_read(f).expect(&format!(
+                "Error decoding lagrange cache file {:?}",
+                cache_key
+            ));
+            println!(
+                "Successully loaded lagrange basis from cache {:?}",
+                cache_key
+            );
+            Some(basis)
+        } else {
+            None
+        }
+    }
+
+    fn cache_lagrange_basis(
+        &self,
+        g: &Vec<G>,
+        domain: &D<G::ScalarField>,
+        basis: &Vec<PolyComm<G>>,
+    ) {
+        let cache_key = self.lagrange_basis_cache_key(g, domain);
+        if Path::exists(&cache_key) {
+            println!("Lagrange basis cache file {:?} already exists", cache_key);
```

Comment:
TODO: Delete

Reply:
will do, these are only here for sanity checking while we're getting set up